tracpro/contacts/fields.py

```diff
@@ -44,70 +56,98 @@ def render(self, name, value, attrs=None):
         return mark_safe(''.join(output))
 
 
-def validate_twitter_handle(handle):
+def check_urns(client, uuid, urn):
+    '''Checking the local Contacts for a urn.
+        If none exist, checking RapidPro for a urn.
+        If none exist, returning False.'''
+    from .models import Contact
+    # RapidPro checks for the numeric id without handle
+    # we are going to mimic that here locally
+
+    if urn.startswith('twitterid'):
+        contacts = Contact.objects.filter(urn__icontains=urn.split('#')[0])
+        contacts = contacts.exclude(uuid=uuid)
+    else:
+        contacts = Contact.objects.filter(urn=urn).exclude(uuid=uuid)
+    if contacts.count() != 0:
+        return True
+    else:
+        if client.check_urn_exists(urn, uuid) is True:
```

Comment:
Noted and will think to refactor as I go.  This `client.check_urn_exists(urn, uuid) is True` is explicitly in place because of testing.  Mocking, specifically, makes it so a test fails with a 200 error with an error stating that the phone number already exists, instead of a 302 as expected.  Even when a phone number does not exist.

Reply:
Since the client is mocked, the mock object won't have a check_urn_exists method on it, and so trying to call that will just return another mock, which will be "true-ish".  If you just want to make check_urn_exists return True or False, you can do something like this first::

    client.check_urn_exists.return_value = False

and then anything calling check_urn_exists on that mock client will get back False.

If you need to test check_urn_exists itself, you'll probably need to use a separate test class that doesn't inherit from the Tracpro test class that always mocks the client. Or create a new client in your test and use that.